test/e2e/autotls/auto_tls_test.go

```diff
@@ -25,89 +25,78 @@ import (
 	"testing"
 	"time"
 
+	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 
-	pkgtest "knative.dev/pkg/test"
 	"knative.dev/serving/pkg/apis/networking"
-	cmclientset "knative.dev/serving/pkg/client/certmanager/clientset/versioned"
 	routenames "knative.dev/serving/pkg/reconciler/route/resources/names"
 	"knative.dev/serving/test"
 	testingress "knative.dev/serving/test/conformance/ingress"
 	"knative.dev/serving/test/e2e"
 	v1test "knative.dev/serving/test/v1"
-
-	"github.com/ghodss/yaml"
-	cmv1alpha2 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
-	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
 )
 
 const (
 	systemNamespace = "knative-serving"
 )
 
-var (
-	caClusterIssuer = &cmv1alpha2.ClusterIssuer{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "ca-issuer",
-		},
-		Spec: cmv1alpha2.IssuerSpec{
-			IssuerConfig: cmv1alpha2.IssuerConfig{
-				CA: &cmv1alpha2.CAIssuer{},
-			},
-		},
-	}
-)
-
-type autoTLSClients struct {
-	clients  *test.Clients
-	cmClient cmclientset.Interface
-}
-
+// To run this test locally with cert-manager, you need to
+// 1. install cert-manager from third_party/.
+// 2. Run below command to do the configuration:
+// kubectl apply -f test/config/autotls/certmanager/selfsigned/
 func TestPerKsvcCert_localCA(t *testing.T) {
-	tlsClients := initializeClients(t)
-	disableNamespaceCert(t, tlsClients)
+	clients := e2e.Setup(t)
```

Comment:
nit, why not `client` instead of `clients`? this returns slice of client?

Reply:
I follow the existing ways https://github.com/knative/serving/blob/master/test/e2e/autoscale_test.go#L65. The `clients` contains multiple client.

Reply:
good to know

Reply:
It is a "set" of clients :)